src/bin/pg_autoctl/string_utils.c

```diff
@@ -443,3 +443,29 @@ splitLines(char *errorMessage, char **linesArray, int size)
 
 	return lineNumber;
 }
+
+
+/*
+ * processBufferCallback is a function callback to use with the subcommands.c
+ * library when we want to output a command's output as it's running, such as
+ * when running a pg_basebackup command.
+ */
+void
+processBufferCallback(const char *buffer, bool error)
+{
+	char *outLines[BUFSIZE] = { 0 };
+	int lineCount = splitLines((char *) buffer, outLines, BUFSIZE);
+	int lineNumber = 0;
+
+	for (lineNumber = 0; lineNumber < lineCount; lineNumber++)
+	{
+		/*
+		 * pg_basebackup and other utilities write their progress output on
+		 * stderr, we don't want to have ERROR message when it's all good.
```

Comment:
This comment seems to have nothing to do with the code below. It seems that simply ignores empty lines. It's good to note somewhere that processBufferCallback is called both for `stderr` and `stdout`, but this does not seem like the right place.

Reply:
Ah yeah. Good point. Revised now, thanks!